### PR TITLE
Fix capitalization issue

### PR DESCRIPTION
**Pull request by benc-db**
### Description

Incremental tables were potentially not running incrementally if components were specified with capitalization, as in information_schema they are always present in lowercase.  Talking to dbt to see if there is a better way to do this, but for now, ensure lower case when used in queries.

### Checklist

- [x] I have run this code in development and it appears to resolve the stated issue
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt-databricks next" section.

### CODE REVIEW

File: dbt/include/databricks/macros/adapters/catalog.sql [hunk 1]
```diff
@@ -69,23 +69,23 @@
...
 {% macro databricks__get_catalog_relations_where_clause_sql(catalog, relations) -%}
-    where table_catalog = '{{ catalog }}' and (
+    where table_catalog = '{{ catalog|lower }}' and (
         {%- for relation in relations -%}
             {% if relation.schema and relation.identifier %}
                 (
-                    table_schema = lower('{{ relation.schema }}')
-                    and table_name = lower('{{ relation.identifier }}')
+                    table_schema = '{{ relation.schema|lower }}'
+                    and table_name = '{{ relation.identifier|lower }}'
```
Comment (jackyhu-db) [inline dbt/include/databricks/macros/adapters/catalog.sql:84]: what if the table name is configured as case-sensitive

Comment (benc-db) [reply dbt/include/databricks/macros/adapters/catalog.sql:84]: Can you link to ref?  This would be a considerable monkey wrench if so.

Comment (jackyhu-db) [reply dbt/include/databricks/macros/adapters/catalog.sql:84]: nvm. Just found Databricks table name is always case insenstive.

**Review by jackyhu-db — DISMISSED** [review]

**Comment by benc-db** [comment]
Going to merge (after changelog) and if dbt has a better option, will do that as a separate PR.

**Pull request merged**